crates/filtering/src/madgwick.rs

```diff
@@ -0,0 +1,94 @@
+use nalgebra::{
+    Matrix4, Quaternion, RealField, Scalar, SimdValue, UnitQuaternion, Vector3, Vector4,
+};
+use num_traits::{One, Zero};
+
+#[derive(Debug)]
+pub struct AccelerometerNormZero;
+
+/// Madgwick's filter for orientation estimation.
+///
+/// This implementation is based on the following:
+/// https://github.com/jmagnuson/ahrs-rs
+pub trait Madgwick<T> {
```

Comment:
I don't know about having `Madgwick` generic. In my opinion it's enough to implement for `UnitQuaternion<f32>` to remove all the generics. In the rest of our codebase we use `f32` anyway

Reply:
it is generic already, and it was from the source I copied it from. Is there any reason to drop this existing implementation?